lib/lambda_ethereum_consensus/p2p/incoming_requests/handler.ex

```diff
@@ -88,8 +99,59 @@ defmodule LambdaEthereumConsensus.P2P.IncomingRequests.Handler do
     end
   end
 
-  defp handle_req("beacon_blocks_by_range/2/ssz_snappy", _message_id, _message),
-    do: :not_implemented
+  defp handle_req("beacon_blocks_by_range/2/ssz_snappy", message_id, message) do
+    with <<24, snappy_blocks_by_range_request::binary>> <- message,
+         {:ok, ssz_blocks_by_range_request} <- Snappy.decompress(snappy_blocks_by_range_request),
+         {:ok, blocks_by_range_request} <-
+           Ssz.from_ssz(ssz_blocks_by_range_request, SszTypes.BeaconBlocksByRangeRequest) do
+      blocks_by_range_request
+      |> inspect(limit: :infinity)
+      |> then(&"[Received BlocksByRange Request] '#{&1}'")
+      |> Logger.debug()
+
+      ## TODO: there should be check that the `start_slot` is not older than the `oldest_slot_with_block`
+      %SszTypes.BeaconBlocksByRangeRequest{start_slot: start_slot, count: count} =
+        blocks_by_range_request
+
+      count =
+        if count > ChainSpec.get("MAX_REQUEST_BLOCKS") do
+          ChainSpec.get("MAX_REQUEST_BLOCKS")
+        else
+          count
+        end
+
+      slot_coverage = start_slot + (count - 1)
+
+      blocks =
+        start_slot..slot_coverage
+        |> Enum.map(fn slot -> BlockStore.get_block_by_slot(slot) end)
+
+      response_chunk =
+        blocks
+        |> Enum.reduce(<<>>, fn block_response, response_chunk ->
```

Comment:
You can use `Enum.map` and then `Enum.join` to simplify this. Also, this:

```elixir
fn a ->
  case a do
    {:ok, ...} -> ...
    {:error, ...} -> ...
  end
end
```

is equivalent to this:

```elixir
fn
  {:ok, ...} -> ...
  {:error, ...} -> ...
end
```